libtransact/src/protocol/receipt.rs

```diff
@@ -49,6 +50,16 @@ impl StateChange {
     }
 }
 
+impl Into<state::StateChange> for StateChange {
```

Comment:
Since `state` and `protocol` live in the same crate, you'd actually want to implement `From` on `state::StateChange` which will implicitly implement `Into` on `receipt::StateChange`

https://doc.rust-lang.org/std/convert/trait.Into.html

Reply:
Can you do that within the receipt module to avoid importing receipt into the state module?

Reply:
I implemented Into instead of From because of this conversation on rocket chat: https://chat.hyperledger.org/channel/transact?msg=aEaCj3J6R5w26m6BJ.

The reasoning was that receipt already does conversions so it should support the conversion and not the other way around, and we only need the conversion one way. We wouldn't need to go from state to receipt.

Please let me know if that is no longer the decision, I can definitely switch to From. As you said we would get the Into for free.

Reply:
You can since they're in the same crate. Personally, I think `From`s should be in the same file as the struct definition, but it's more than valid.

Reply:
However, if the intention is to have protocol be it's own crate eventually then I think `Into` is fine.

Reply:
I didn't realize you could implement From in the receipt module. Just to clarify then, which trait would we like?

Reply:
It should be

```
impl From<StateChange> for receipt::StateChange {
...
}
```

Reply:
Great, updated to implement the `From` trait. Thanks!

Reply:
So I just had a spirited discussion with @vaporos, and we came to the conclusion that `Into` would actually be the correct trait since state shouldn't be aware of protocol. I'm really sorry for the inconvenience/ confusion.

Reply:
Okay, no problem. Not a difficult change to make :). Updated the PR.

Reply:
For context - the reason we think Into is the right thing here, is that From would add to the state::State API, which we were explicitly trying to keep "clean" of receipt. Even if the From is implemented in the receipt module, the API of state::State changes. Thus the reasoning behind keeping it only Into.